src/main/java/com/onarandombox/MultiversePortals/commands/ListCommand.java

```diff
@@ -100,12 +89,43 @@ public void runCommand(CommandSender sender, List<String> args) {
 
     private List<String> getPortals(CommandSender sender, MultiverseWorld world, String filter) {
         List<String> portals = new ArrayList<>();
-        if (filter == null)
+        if (filter == null) {
             filter = "";
-        boolean altColor = false;
+        }
         for (MVPortal p : (world == null) ? this.plugin.getPortalManager().getPortals(sender) : this.plugin.getPortalManager().getPortals(sender, world)) {
-            if (p.getName().matches("(i?).*" + filter + ".*"))
-                portals.add(ChatColor.YELLOW + p.getName() + ((p.getDestination() != null) ? (ChatColor.AQUA + " -> " + ChatColor.GOLD + p.getDestination().getName()) : ""));
+            String destination = "";
+            if(p.getDestination() != null) {
+                destination = p.getDestination().toString();
+
+                MultiverseWorld destWorld = this.plugin.getCore().getMVWorldManager().getMVWorld(destination);
+                if (destWorld != null) {
+                    destination = "(World) " + ChatColor.DARK_AQUA + destination;
+                } else {
+                    String destType = destination.substring(0, 1);
+                    if (destType.equals("p")) {
+                        destination = "(Portal) " + ChatColor.DARK_AQUA + p.getDestination().getName();
+                    }
+                    if (destType.equals("e")) {
+                        //String destParts = p.getDestination().toString().substring(2, p.getDestination().toString().length()-1);
+                        String destinationWorld = p.getDestination().toString().split(":")[1];
+                        String destPart = p.getDestination().toString().split(":")[2];
+                        String[] locParts = destPart.split(",");
+                        int x, y, z;
+                        try {
+                            x = (int) Double.parseDouble(locParts[0]);
+                            y = (int) Double.parseDouble(locParts[1]);
+                            z = (int) Double.parseDouble(locParts[2]);
+                        } catch(NumberFormatException e) {
+                            e.printStackTrace();
+                            continue;
+                        }
+                        destination = "(Location) " + ChatColor.DARK_AQUA + destinationWorld + ", " + x + ", " + y + ", " + z;
+                    }
```

Comment:
Instead of the way you did with `destWorld`, you could do something like this, since world destination actually has an identifier `w`
```java
String destType = p.getDestination().getIdentifier();
if (destType =="w") {
    // nice format for world
}
else if (destType =="p") {
    // nice format for portal
}
else if (destType =="e") {
    // nice format for exact dest
}
```